dev/dll/Microsoft.UI.Xaml.vcxproj

```diff
@@ -239,6 +243,12 @@
       <PreprocessorDefinitions>DISABLE_WINRT_DEPRECATION;%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <Optimization Condition="'$(Configuration)'=='Release'">MinSpace</Optimization>
       <ConformanceMode>true</ConformanceMode>
+      <!-- Pipelines agents are low-memory configurations that struggle to compile code analysis in parallel. See 
+          https://developercommunity.visualstudio.com/content/problem/831568/cl-analyze-runs-out-of-memory.html
+          for more details. While we get that resolved (either with a fix in VS or better agents), we need
+          to disable MultiProc compile. Use NUMBER_OF_PROCESSORS==2 to check if it's one of the underpowered Azure
+          Pipelines VMs -->
+      <MultiProcessorCompilation Condition="'$(Configuration)'=='Debug' AND '$(NUMBER_OF_PROCESSORS)'=='2'">false</MultiProcessorCompilation>
```

Comment:
Should this condition be expressed by:
RunCodeAnalysis=True && TF_BUILD==True
?
That seems to better capture the intent here.

Reply:
Agree with checking $(RunCodeAnalysis) instead of Configuration==Debug, but I think checking NUMBER_OF_PROCESSORS is better than TF_BUILD because it handles the case that our pipelines machines get better specs without us doing anything, the perf just gets better. But it assumes that when they get more procs they also get more memory, so it's not perfect. I could go either way -- what do you think?